File: java/spring-resilience4j/spring-resilience4j-api-a/src/main/java/com/bmarques/springresilience4japia/revenue/RevenueController.java
package com.bmarques.springresilience4japia.revenue;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/revenue")
public class RevenueController {

    @Autowired
    private RevenueService service;

    @GetMapping
    public Double getTotalRevenue() {
        return service.getTotalRevenue();
    }
}

File: kotlin/spring-resilience4j/spring-resilience4j-api-a/src/main/kotlin/com/bmarques/springresilience4japia/revenue/RevenueController.kt
package com.bmarques.springresilience4japia.revenue

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/revenue")
class RevenueController(private val revenueService: RevenueService) {

    @GetMapping
    fun getTotalRevenue() = revenueService.getTotalRevenue()

}